phpunit/functional/AuthLdapTest.php

```diff
@@ -0,0 +1,198 @@
+        $this->checkAuthIsNotDefault($changingAuthLDAP);
+        $this->checkAuthIsActive($changingAuthLDAP);
+
+        // Act - update an existing AuthLdap as default
+        $updated = $this->updateItem(AuthLDAP::class, $changingAuthLDAP->getID(), ['is_default' => 1]);
+
+        // Assert
+        $this->assertDefaultAuthIs($updated);
+    }
+
+    public function test_UpdateAuthLdapDoesntChangeDefaultAuthLdap()
+    {
+        // Arrange - ensure there is one default AuthLdap
+
+        /** @var AuthLDAP $changingAuthLDAP */
+        $changingAuthLDAP = getItemByTypeName(AuthLDAP::class, 'LDAP3', throw_exception: true);
+
+        $this->checkAuthsAreDifferent($this->initialDefaultAuth, $changingAuthLDAP);
+        $this->checkAuthClassesAreTheSame($this->initialDefaultAuth, $changingAuthLDAP);
+        $this->checkAuthIsNotDefault($changingAuthLDAP);
+        $this->checkAuthIsActive($changingAuthLDAP);
+
+        // Act - update an existing AuthLdap not as default
+        $this->updateItem(AuthLDAP::class, $changingAuthLDAP->getID(), ['is_default' => 0]);
+
+        // Assert
+        $this->assertDefaultAuthUnchanged();
+    }
+
+    // --- interactions with AuthMail
+
+    public function test_AddNonDefaultAuthMailDoesntChangeDefaultAuthLdap()
+    {
+        // Arrange - Ensure default Auth is an AuthLDAP
+
+        // Act - add a AuthLDAP as default
+        $this->createItem(AuthMail::class, [
+            'is_default' => 0,
+            'name' => 'newmail',
+            'is_active' => 1,
+        ]);
+
+        // Assert
+        $this->assertDefaultAuthUnchanged();
+    }
+
+    public function test_AddDefaultAuthMailChangeDefaultAuth()
+    {
+        // Act - add an AuthLDAP as default
+        $created = $this->createItem(AuthLDAP::class, [
+            'is_default' => 1,
+            'name' => 'newldap',
+            'is_active' => 1,
+        ]);
+
+        // Assert
+        $this->assertDefaultAuthIs($created);
+    }
+
+    public function test_UpdateAuthMailToDefaultChangeDefaultAuthLdap()
+    {
+        // Arrange
+        $authMail = getItemByTypeName(AuthMail::class, 'MAIL3', throw_exception: true);
+        $this->checkAuthIsNotDefault($authMail);
+        $this->checkAuthIsActive($authMail);
+
+        $initialDefaultAuthLDAP = $this->getDefaultAuth(AuthLDAP::class); // must be after $authMail creation
+        $this->checkAuthIsActive($initialDefaultAuthLDAP);
+
+        // Act
+        $updated = $this->updateItem(AuthMail::class, $authMail->getID(), ['is_default' => 1, 'name' => 'MAIL2']); // name is mandatory
+
+        // Assert
+        $this->assertDefaultAuthIs($updated);
+    }
+
+    private function createDefaultAuthLdap()
+    {
+        // remove ldap from loadDataset(), make our independent from initial dataset
+        $query = "DELETE FROM " . AuthLDAP::getTable();
+        global $DB;
+        assert($DB->doQuery($query), 'Failed to empty AuthLDAP table');
```

Comment:
Usage of `$DB->doQuery()` should be avoid in new code.

```suggestion
        global $DB;
        assert($DB->->delete(AuthLDAP::getTable(), [1]), 'Failed to empty AuthLDAP table');
```